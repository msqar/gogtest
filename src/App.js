import React from 'react';
import Header from './components/Header/Header';
import HomeView from './views/HomeView/HomeView';

class App extends React.Component {
    render() {
        return (
            <div className="App">
                <Header></Header>
                <HomeView></HomeView>
            </div>
        );
    }
}

export default App;
