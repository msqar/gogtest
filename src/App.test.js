// import { render, screen } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders app without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

it('shallow renders without crashing', () => {
    shallow(<App />);
});

it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});
