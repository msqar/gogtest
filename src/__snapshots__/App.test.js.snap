// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="Header navbar navbar-expand navbar-light fixed-top"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt="GOG"
          className="Header-logo d-inline-block align-top"
          src="/images/gog_logo.png"
        />
      </a>
      <div
        className="Cart "
      >
        <button
          className="Cart-button"
          onClick={[Function]}
        >
          <img
            alt="Cart_Icon"
            className="Cart-icon"
            src="cart_icon.svg"
          />
          <span
            className="Cart-counter"
          >
            0
          </span>
        </button>
        <div
          className="Cart-dropdown "
        >
          <div
            className="Cart-dropdownBody is--empty"
          >
            <img
              alt="Cart_Icon"
              className="Cart-dropdown-cartIcon"
              src="cart_icon.svg"
            />
            <span
              className="Cart-dropdown-emptyTitle"
            >
              Your Cart is Empty
            </span>
            <div
              className="Divider"
            />
            <span
              className="Text--secondary Text--14"
            >
              Explore great games and offers
            </span>
            <button
              className="Button--secondary u-marginTl"
            >
              BROWSE BESTSELLERS
            </button>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container-md"
  >
    <div
      className="OverlayBackground "
      onClick={[Function]}
    />
    <div
      className="HomeView"
    >
      <section
        className="Section"
      >
        <span
          className="Section-title text-uppercase"
        >
          GAME OF THE WEEK
        </span>
        <div
          className="Banner"
        >
          <div
            className="Banner-imageContainer"
          >
            <img
              alt="highlight_game"
              className="Banner-image"
              src="images/game_bg_xl.png"
            />
          </div>
          <div
            className="Section-gameResults"
          >
            <div
              className="row"
            />
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
